### PR TITLE
@typescript-eslint/no-namespace

### PR DESCRIPTION
**Pull request by metreniuk**
**Overview**
Fixes the `no-namespace` rule for https://github.com/web-infra-dev/oxc/issues/503

**Help needed**
- I'm not sure how to better adjust the `Tester` to support running tests with different extensions, so I did the bare minimum and would appreciate guidance how to make it actually good. My current approach is just running a separate test/snapshot for tests that require a different extension. I was thinking adding it inside the options `json`, but it seemed unintuitive to use.
- I removed the guard around typescript declarations inside `crates/oxc_semantic/src/builder.rs`, but I'm not sure why it was in the first place.

**Notes**
- Added a `typescript-eslint` rule generator. Didn't support `filename` since had trouble adjusting the `Tester` module.

### CODE REVIEW

**Comment by Boshen** [comment]
We don't need to lint `.d.ts` because they are either machine generated or manually crafted.

**Comment by Boshen** [comment]
Since the `namespace` syntax works on ordinary TypeScript files, perhaps we don't need to change anything around filenames?

**Comment by metreniuk** [comment]
> Since the `namespace` syntax works on ordinary TypeScript files, perhaps we don't need to change anything around filenames?

In the original rule there is an option `allowDefinitionFiles`, if we don't lint those I can just remove the option. Wdyt?

**Comment by Boshen** [comment]
> 

Let's keep the parsed value and , we just don't use it for now. 

> > Since the `namespace` syntax works on ordinary TypeScript files, perhaps we don't need to change anything around filenames?
> 
> In the original rule there is an option `allowDefinitionFiles` that allows namespaces in `d.ts`, if we don't lint those I can just remove the option. Wdyt?

Sounds good to me! Maybe keep the `allowDefinitionFiles` value in the code but comment out the relevant test cases?

File: crates/oxc_semantic/src/builder.rs [hunk 1]
```diff
@@ -126,9 +126,7 @@ impl<'a> SemanticBuilder<'a> {
 
     pub fn build(mut self, program: &'a Program<'a>) -> SemanticBuilderReturn<'a> {
         // First AST pass
-        if !self.source_type.is_typescript_definition() {
-            self.visit_program(program);
-        }
```
Comment (Boshen) [inline crates/oxc_semantic/src/builder.rs:131]: Need to move this code back 😁

**Review by Boshen — APPROVED** [review]

**Comment by Boshen** [comment]
Thank you! Love the addition to rule gen!

**Pull request merged**